src/WooCommerce.php

```diff
@@ -389,14 +397,25 @@ public static function woocommerce_save_product_variation($post_id, $loop) {
     foreach ($custom_fields as $field) {
       if (isset($_POST[$field]) && isset($_POST[$field][$loop])) {
         if ($_POST[$field][$loop]) {
-          update_post_meta($variation_id, $field, $_POST[$field][$loop]);
+          if ($field !== '_' . Plugin::PREFIX . '_gtin') {
+            update_post_meta($post_id, $field, $_POST[$field]);
+          } else if (!self::is_existing_gtin($variation_id, $_POST[$field][$loop])) {
+            // GTIN is only saved if unique.
+            update_post_meta($variation_id, $field, $_POST[$field][$loop]);
+          } else {
+            $gtin_error = true;
+          }
```

Comment:
I'd propose to refactoring the conditional logic to ensure the GTIN error message is set only in case the field is GTIN and the GTIN is existing.

Something like what I proposed above.